confluent_kafka/serialization/serializers.py

```diff
@@ -0,0 +1,455 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+# Copyright 2020 Confluent Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import struct
+
+from .error import SerializerError
+
+
+class SerializerBase(object):
+    """
+    Extensible class which all Serializer implementations must derive from.
+
+    Serializers instruct kafka clients on how convert Python objects to bytes.
+
+    Note:
+        This class is not directly instantiable. The derived classes should be
+        used instead.
+
+    The following implementations are provided by this module.
+
+    .. list-table::
+        :header-rows: 1
+
+        * - Name
+          - Python Type
+          - C Type(bytes)
+
+        * - DoubleSerializer
+          - float
+          - double
+        * - ShortSerializer
+          - int
+          - short
+        * - LongSerializer
+          - int
+          - long long
+        * - IntegerSerializer
+          - int
+          - int(signed)
+        * - StringSerializer
+          - unicode
+          - bytes
+
+    Unless noted elsewhere all serialized bytes are stored as big-endian.
+
+    """
+    def serialize(self, datum, ctx):
+        """
+        Converts datum to bytes.
+
+        :param object datum: float to be encoded
+        :param SerializationContext ctx: Serialization metadata
+
+        :raises: SerializerError
+
+        :returns: C type double bytes
+        :rtype: bytes
+        """
+        raise NotImplementedError()
+
+    def deserialize(self, data, ctx):
+        """
+        Convert bytes to object
+
+        :param bytes data: bytes to be decoded
+        :param SerializationContext ctx:
+
+        :raises: SerializerError
+
+        :returns: decoded float
+        :rtype: float
+        """
+        raise NotImplementedError()
+
+
+class DoubleSerializer(SerializerBase):
+    """
+    Apache Kafka conformant serializer/deserializer.
+    See the javadocs for additional details:
+    https://kafka.apache.org/20/javadoc/org/apache/kafka/common/serialization/DoubleSerializer.html
+    https://kafka.apache.org/20/javadoc/org/apache/kafka/common/serialization/DoubleDeserializer.html
+    """
+
+    def __init__(self):
+        pass
+
+    @staticmethod
+    def serialize(datum, ctx):
```

Comment:
this class doesn't provide implementations for the methods in `SerializerBase` - static methods wouldn't override the base class implementations, they'd provide methods on the class object. So this seems misleading/confused to me. just rely on duck typing.

[usual disclaimer about not knowing python applies]

Reply:
Yeah that was just me taking advantage if pythons method resolution semantics. From the receivers perspectice there really is no difference but you're right its a hack.